src/types.ts

```diff
@@ -85,22 +84,32 @@ export type Breadcrumbs = {
   [key: string]: Crumb;
 };
 
-export interface EnrichedCrumb extends Crumb {
+export interface NormalizedCrumb extends Crumb {
   id: string;
+  parentId: string | null; // null if it is the first node
   autoAnswered: boolean;
   sectionId?: string;
-  details: {
-    component: string;
-    nodeData?: Record<string, Value>;
-    answerData?: {
-      [id: NodeId]: Record<string, Value>;
-    };
-  };
+  rootNodeId: string;
 }
```

Comment:
Enough information has been added to allow an array of `NormalizedCrumb`s to be easily traversed for queries but the data from each node can be looked-up separately (that's why this has changed from `EnrichedCrumb`).

Reply:
sorry to be catching up here after this merged, but a couple questions coming up for me as I'm reading thorugh these changes:
- what exactly is `rootNodeId` here & on `NormalizedFlow`?
- how is it different than `id` ?